File: laptop/Python_Solution/단계별/1차원배열/4344.py
# Over-Average

from sys import stdin, stdout
inp = stdin.readline
pr = stdout.write

class OverAverageApp:
    def __init__(self):
        self.ans = 0.0
        self.getScoreInput()
        
    def getScoreInput(self):
        s = list(map(float, inp().split(" ")))
        n, scoreList = s[0], s[1:]
        self.getAvg(n, scoreList)
    
    def getAvg(self, n, scoreList):
        avg = sum(scoreList) / n
        self.checkToPrint(n, avg, scoreList)
    
    def checkToPrint(self, n, avg, scoreList):
        you = 0
        for i in scoreList:
            if i > avg:
               you += 1
        self.ans = "%.3f" % ((float(you) / n) * 100.0)
    

if __name__ == "__main__":
    ansList = []
    for _ in range(int(inp())):
        oa = OverAverageApp()
        ansList.append(oa.ans+"%")
    pr('\n'.join(ansList))

File: laptop/C_Solution/단계별/1차원배열/4344.c
// Over-Average

#include <stdio.h>

void checkToPrint(double avg, double list[], double n){
    int you = 0;
    for(int i = 0; i < (int)n; i++){
        if(list[i] > avg){
            you++;
        }
    }
    printf("%.3lf%%\n", ((double)you / n) * 100.0);
}

int main(void){
    int testCaseCnt;
    scanf("%d", &testCaseCnt);
    char c = getchar(); // flushing buffer

    for(int i = 0; i < testCaseCnt; i++){
        double scoreList[1000] = {0, };
        double n, tmpScore=0.0;
        scanf("%lf", &n);
        for(int j = 0; j < (int)n; j++){
            scanf("%lf", &scoreList[j]);
            tmpScore += scoreList[j];
        }
        double avg = tmpScore / n;
        checkToPrint(avg, scoreList, n);
    }
    return 0;
}